WooCommerce/Classes/POS/Controllers/PointOfSaleOrderController.swift

```diff
@@ -0,0 +1,145 @@
+    }
+
+    private let orderService: POSOrderServiceProtocol
+
+    private let currencyFormatter: CurrencyFormatter
+
+    @Published private var orderState: PointOfSaleInternalOrderState = .idle
+    private(set) var order: Order? = nil
+
+    @MainActor
+    func syncOrder(for cartProducts: [CartItem],
+                   retryHandler: @escaping () async -> Void) async {
+        guard !orderState.isSyncing,
+              CartItem.areOrderAndCartDifferent(order: order, cartItems: cartProducts) else {
+            return
+        }
+
+        orderState = .syncing
+        let cartItems = cartProducts.map {
+            POSCartItem(product: $0.item, quantity: Decimal($0.quantity))
+        }
+
+        do {
+            let syncedOrder = try await orderService.syncOrder(cart: cartItems, order: order)
+            self.order = syncedOrder
+            orderState = .loaded(totals(for: syncedOrder), syncedOrder)
+            DDLogInfo("🟢 [POS] Synced order: \(syncedOrder)")
+        } catch {
+            DDLogError("🔴 [POS] Error syncing order: \(error)")
+            setOrderStateToError(error, retryHandler: retryHandler)
+        }
+    }
+
+    private func setOrderStateToError(_ error: Error,
+                                      retryHandler: @escaping () async -> Void) {
+        // Consider removing error or handle specific errors with our own formatting and localization
+        orderState = .error(.init(message: error.localizedDescription,
+                                  handler: {
+            Task {
+                await retryHandler()
+            }
+        }))
+    }
+
+    func clearOrder() {
+        order = nil
+        orderState = .idle
+    }
+}
+
+
+private extension PointOfSaleOrderController {
+    func totals(for order: Order) -> PointOfSaleOrderTotals {
+        let totalsCalculator = OrderTotalsCalculator(for: order,
+                                                     using: currencyFormatter)
+        return PointOfSaleOrderTotals(
+            cartTotal: formattedPrice(totalsCalculator.itemsTotal.stringValue,
+                                      currency: order.currency) ?? "",
+            orderTotal: formattedPrice(order.total, currency: order.currency) ?? "",
+            taxTotal: formattedPrice(order.totalTax, currency: order.currency) ?? "")
+    }
+
+    func formattedPrice(_ price: String?, currency: String?) -> String? {
+        guard let price, let currency else {
+            return nil
+        }
+        return currencyFormatter.formatAmount(price, with: currency)
+    }
+}
+
+// This is named to note that it is for use within the AggregateModel and OrderController.
+// Conversely, PointOfSaleOrderState is available to the Views, as it doesn't include the Order.
+enum PointOfSaleInternalOrderState {
+    case idle
+    case syncing
+    case loaded(PointOfSaleOrderTotals, Order)
+    case error(PointOfSaleOrderSyncErrorMessageViewModel)
+
+    var isSyncing: Bool {
```

Comment:
It could be, but it would be reasonable for the aggregate model to check this too; it's simply a more convenient way to check `if case .syncing = orderState`, especially when you need to know the negative.